### PR TITLE
call OpenAI from the CLI 

### PR DESCRIPTION
**Pull request by OceanOak**
Changelog:

```
Cli
- Enable the Cli to interact with the OpenAI API. 
```
This will enable users to send prompts through the CLI and save the generated output into a specified script

### CODE REVIEW

File: canvases/dark-packages/main.dark [hunk 1]
```diff
@@ -254,4 +254,12 @@ let _handler _req =
     | Some Module -> "module"
     | None -> "not found"
 
-  PACKAGE.Darklang.Stdlib.Http.response (respBody |> String.toBytes) 200
\ No newline at end of file
+  PACKAGE.Darklang.Stdlib.Http.response (respBody |> String.toBytes) 200
+
+
+
+[<HttpHandler("GET", "/openai-chat/:prompt")>]
```
Comment (OceanOak) [inline canvases/dark-packages/main.dark:261]: It seems like this HTTP handler doesn't belong in this canvas. Maybe we should move it to a separate canvas?

Comment (StachuDotNet) [reply canvases/dark-packages/main.dark:261]: I'm not sure why we need this endpoint - why not do the `(PACKAGE.OpenAI.ChatCompletion.completion prompt) |> unwrap` in the CLI code?

Comment (OceanOak) [reply canvases/dark-packages/main.dark:261]: In the CLI, we use `CLI.HttpClient.request`, while `PACKAGE.OpenAI.ChatCompletion.completion` uses `HttpClient.request`. Attempting to use it directly results in a 'Function HttpClient.request is not found' error. This is why I needed to add the endpoint.

File: canvases/dark-editor/system-prompt-cli.txt [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+Here are some examples of a response to the following prompt:
+
+prompt: build a small program that models a list of People,
+a function that returns the age of a Person, then hardcode a list of people
+and return the age of each person
+
+Response:
+
+type Person = {
+name: String
```
Comment (pbiggar) [inline canvases/dark-editor/system-prompt-cli.txt:131]: I think some spacing got missed here.

File: canvases/dark-packages/main.dark (hunk 1 above)
Comment (pbiggar) [reply canvases/dark-packages/main.dark:261]: Yes, i agree with Stachu. I guess we need to detangle HttpClient.request from LibBackend.

File: packages/darklang/cli/cli.dark [hunk 3]
```diff
@@ -246,19 +246,63 @@ module Darklang =
         1
 
 
+    let generateCode (prompt: String) (scriptPath: String) : Int =
+
+      let prompt =
+        match File.read "canvases/dark-editor/system-prompt-cli.txt" with
```
Comment (pbiggar) [inline packages/darklang/cli/cli.dark:252]: The prompt should be a constant.

**Comment by pbiggar** [comment]
There's a few things I'd change, but no reason to hold this up. I'll make a few more comments inline.

File: canvases/dark-editor/system-prompt-cli.txt (hunk 2 above)
Comment (pbiggar) [reply canvases/dark-editor/system-prompt-cli.txt:131]: Maybe throughout this whole file in fact.

File: packages/darklang/cli/cli.dark [hunk 4]
```diff
@@ -246,19 +246,63 @@ module Darklang =
...
+        | Ok darkPrompt -> prompt ++ "\n" ++ (darkPrompt |> String.fromBytes)
+        | Error e ->
+          print e
+          prompt
+
+      print prompt
+
+      let request =
```
Comment (pbiggar) [inline packages/darklang/cli/cli.dark:260]: I would call this `response`

**Pull request merged**